src/command_modules/azure-cli-batch/azure/cli/command_modules/batch/custom.py

```diff
@@ -0,0 +1,96 @@
+# --------------------------------------------------------------------------------------------
+
+try:
+    from urllib.parse import urlsplit
+except ImportError:
+    from urlparse import urlsplit # pylint: disable=import-error
+
+from azure.mgmt.batch.models import (BatchAccountCreateParameters,
+                                     AutoStorageBaseProperties,
+                                     UpdateApplicationParameters)
+
+from azure.mgmt.batch.operations import (ApplicationPackageOperations)
+import azure.cli.core._logging as _logging
+from azure.storage.blob import BlockBlobService
+
+logger = _logging.get_az_logger(__name__)
+
+def list_accounts(client, resource_group_name=None):
+    acct_list = client.list_by_resource_group(resource_group_name=resource_group_name) \
+        if resource_group_name else client.list()
+    return list(acct_list)
+
+def create_account(client, resource_group_name, account_name, location, #pylint:disable=too-many-arguments
+                   tags=None, storage_account_id=None):
+    # TODO: get storage_account_id by search storage account name
+    if storage_account_id:
+        properties = AutoStorageBaseProperties(storage_account_id=storage_account_id)
+    else:
+        properties = None
+
+    parameters = BatchAccountCreateParameters(location=location,
+                                              tags=tags,
+                                              auto_storage=properties)
+
+    return client.create(resource_group_name=resource_group_name,
+                         account_name=account_name,
+                         parameters=parameters)
+create_account.__doc__ = AutoStorageBaseProperties.__doc__
+
+def update_account(client, resource_group_name, account_name,  #pylint:disable=too-many-arguments
+                   tags=None, storage_account_id=None):
+    if storage_account_id:
+        properties = AutoStorageBaseProperties(storage_account_id=storage_account_id)
+    else:
+        properties = None
+
+    return client.update(resource_group_name=resource_group_name,
+                         account_name=account_name,
+                         tags=tags,
+                         auto_storage=properties)
+update_account.__doc__ = AutoStorageBaseProperties.__doc__
+
+def update_application(client, resource_group_name, account_name, application_id, #pylint:disable=too-many-arguments
+                       allow_updates=None, display_name=None, default_version=None):
+    parameters = UpdateApplicationParameters(allow_updates=allow_updates,
+                                             display_name=display_name,
+                                             default_version=default_version)
+    return client.update(resource_group_name=resource_group_name,
+                         account_name=account_name,
+                         application_id=application_id,
+                         parameters=parameters)
+update_application.__doc__ = UpdateApplicationParameters.__doc__
```

Comment:
please add a blank line between this statement and the end of the function definition. additionally, it may be a good idea to write a [decorator](http://thecodeship.com/patterns/guide-to-python-function-decorators/) to accomplish the function like this (assign proxy function's docstring)